include/deal.II/matrix_free/dof_info.h

```diff
@@ -59,6 +58,12 @@ namespace internal
 {
   namespace MatrixFreeFunctions
   {
+    /**
+     * Type of the 8-bit representation of the refinement configuration that
+     * is in hanging_nodes_internal.h.
+     */
+    using compressed_constraint_kind = std::uint8_t;
```

Comment:
Yes, that is unfortunate. I don't see a simple solution for this, since neither `dof_info.h` should pull in all of `hanging_nodes_internal.h` nor the other way around. The good solution would be to separate the declarations and definitions of `hanging_nodes_internal.h`, which would then reduce the interaction. Or a separate header file for the basic types.

Reply:
Would it help to move `ConstraintKinds` to a new file and have a new entry in it: `invalid`?

Reply:
Yes, that is the clean solution, but I didn't want to make such large a change now. (Irrespective of whether we merge this or not, I think it can wait as it is mainly to our own convenience.)